optimism/src/mips/constraints.rs

```diff
@@ -440,10 +450,155 @@ impl<Fp: Field> InterpreterEnv for Env<Fp> {
         _len: &Self::Variable,
         pos: Self::Position,
     ) -> Self::Variable {
-        Expr::Atom(ExprInner::Cell(Variable {
+        // The (at most) 4-byte chunk that has been read from the preimage
+        let bytes: [_; MIPS_CHUNK_BYTES_LENGTH] = array::from_fn(|i| {
+            Expr::Atom(ExprInner::Cell(Variable {
+                col: Self::Position::ScratchState(MIPS_PREIMAGE_BYTES_OFFSET + i),
+                row: CurrOrNext::Curr,
+            }))
+        });
+        // Whether the preimage chunk read has at least n bytes (1, 2, 3, or 4)
```

Comment:
Can you explain the role of these columns?

Reply:
Ah, is it for the alignment?

Reply:
It is basically for the lookup syscall. The idea was that the mips side would create 1 lookup for each byte read from the preimage. But because this could be any number between 1 and 4 bytes in each call, we needed to customize the "structure" of those lookups. For this reason, I created the `has_n_bytes` columns to keep track of whether this instruction had read at least `n` bytes. For example, if the syscall read 3 bytes this time, that means that it has 1 byte, also has 2 bytes, and also has 3 bytes, but not 4 bytes. This means that the lookups that need to be created are the ones corresponding to byte 1, byte 2, and byte 3, leaving the one for the 4th byte set to zero (because `has_n_bytes[3]` will be 0 used in the condition for the write interface). I guess this will have to change when we create different instances for each opcode, we will have to think about how to adapt it.